File: moseoh/_4주차/H_Index/java/Solution.java
package _4주차.H_Index.java;

import java.util.Arrays;

class Solution {
    public static void main(String[] args) {
        int[] citations1 = {3, 0, 6, 1, 5};
        assert new Solution().solution(citations1) == 3 : "실패";
    }

    public int solution(int[] citations) {
        Arrays.sort(citations);

        int max = 0;
        for (int i = citations.length - 1; i > -1; i--) {
            int min = Math.min(citations[i], citations.length - i);
            if (max < min) max = min;
        }

        return max;
    }
}

File: moseoh/_4주차/H_Index/kotlin/Solution.kt
package _4주차.H_Index.H_Index

fun main() {
    val citations1 = intArrayOf(3, 0, 6, 1, 5)
    check(Solution().solution(citations1) == 3) { "오답" }
}

class Solution {
    fun solution(citations: IntArray): Int {
        var answer = 0
        citations.sortDescending()

        for (i in citations.indices) {
            if (i + 1 > citations[i]) break
            answer = i + 1
        }

        return answer
    }
}